OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -88,21 +88,16 @@ void MocoImpulseTrackingGoal::addScaleFactor(const std::string& name,
         "Contact group associated with external force '{}' not found.",
         externalForceName);
 
-    // Check that the index is in the correct range.
-    OPENSIM_THROW_IF_FRMOBJ((index < 0) || (index > 2), Exception,
-        "Expected marker scale factor index to be in the range [0, 2], "
-        "but received '{}'.", index);
-
     // Update the scale factor map so we can retrieve the correct MocoScaleFactor
     // for this contact group during initialization.
-    std::pair<std::string, int> key(externalForceName, index);
+    std::pair<std::string, int> key(externalForceName, impulse_axis);
```

Comment:
Per the comment above, you don't need the `impulse_axis` as part of key now.

Reply:
If I omit `impulse_axis` from here and the declaration/definition of `addScaleFactor`, `key` becomes a problem as it is expecting an integer (due to `pair`) - could this just be initialised with 0? 

Similarly for line 203 - `std::pair<std::string, int> key(group.get_external_force_name(), get_impulse_axis());` to `std::pair<std::string, int> key(group.get_external_force_name(), 0);`

I don't think it should affect anything further down stream...

Reply:
Oh, you can just change `m_scaleFactorMap` ([here](https://github.com/opensim-org/opensim-core/blob/b52ed03f30a8132f449687d3b4db3cc82428a3e5/OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.h#L267)) to the following type:
`mutable std::map<std::string, std::string> m_scaleFactorMap;`